src/modapi/server/modcreator.cpp

```diff
@@ -243,6 +243,103 @@ CModAPI_ModCreator::CModAPI_LineStyleCreator& CModAPI_ModCreator::AddLineStyle()
 	return LineStyle;
 }
 
+CModAPI_ModCreator::CModAPI_SkinModifierCreator& CModAPI_ModCreator::AddSkinModifier()
+{
+	int Id = m_SkinModifiers.size();
+	m_SkinModifiers.set_size(m_SkinModifiers.size()+1);
+	CModAPI_ModCreator::CModAPI_SkinModifierCreator& SkinModifier = m_SkinModifiers[Id];
+
+	// general
+	SkinModifier.m_Id = Id;
+
+	SkinModifier.m_SkinFlag = SKIN_INVALIDMOD;
+	SkinModifier.m_ExternalSet = 0;
+
+	// body
+	SkinModifier.m_BodyFlags = PART_KEEP;
+	SkinModifier.m_BodyColor = 0;
+	SkinModifier.m_BodyFile = 0;
+
+	// marking
+	SkinModifier.m_MarkingFlags = PART_KEEP;
+	SkinModifier.m_MarkingColor = 0;
+	SkinModifier.m_MarkingFile = 0;
+
+	// decoration
+	SkinModifier.m_DecoFlags = PART_KEEP;
+	SkinModifier.m_DecoColor = 0;
+	SkinModifier.m_DecoFile = 0;
+
+	// hands
+	SkinModifier.m_HandsFlags = PART_KEEP;
+	SkinModifier.m_HandsColor = 0;
+	SkinModifier.m_HandsFile = 0;
+
+	// feet
+	SkinModifier.m_FeetFlags = PART_KEEP;
+	SkinModifier.m_FeetColor = 0;
+	SkinModifier.m_FeetFile = 0;
+
+	// eyes
+	SkinModifier.m_EyesFlags = PART_KEEP;
+	SkinModifier.m_EyesColor = 0;
+	SkinModifier.m_EyesFile = 0;
+
+	SkinModifier.m_HookStyle; // a line style ID
+
+	return SkinModifier;
+}
+
+CModAPI_ModCreator::CModAPI_SkinModifierCreator& CModAPI_ModCreator::CModAPI_SkinModifierCreator::SetModifier(int Type, bool UseExternal)
+{
+	m_SkinFlag = Type;
+	m_ExternalSet = UseExternal;
+}
+
+CModAPI_ModCreator::CModAPI_SkinModifierCreator& CModAPI_ModCreator::CModAPI_SkinModifierCreator::SetBody(int Flags, const vec4& Color, const char *File)
+{
+	m_BodyFlags = Flags;
+	m_BodyColor = ModAPI_ColorToInt(Color);
+	m_BodyFile = File;
+}
+
```

Comment:
I can't access the `m_Images` from inside my functions for checking if the `ImageId` is in bounds…